lib/client/access-token.js

```diff
@@ -12,7 +14,15 @@ module.exports = function (config) {
 
   function AccessToken(token) {
     this.token = token;
-    this.token.expires_at = addSeconds(new Date(), token.expires_in);
+    if ({}.hasOwnProperty.call(this.token, 'expires_at')) {
+      if (!isDate(this.token.expires_at)) {
+        this.token.expires_at = parse(this.token.expires_at);
+      }
+    } else {
+      this.token.expires_at = addSeconds(new Date(), token.expires_in);
+    }
+
+    return this;
```

Comment:
Hi @sankethkatta sorry for the late response. Yes, we need to remove that property since now the object is created by using a constructor function. The other change I would suggest is to use `'expires_at' in this.token` instead of calling the hasOwnProperty function.